Exercise_4.java

```diff
@@ -34,7 +34,33 @@ static void insert(Node temp, int key)
     { 
 
         // Do level order traversal until we find 
-        // an empty place and add the node.  
+        // an empty place and add the node. 
+        if(temp==null){

```

Comment:
Good Job! As mentioned previously, please add comments and time and space complexities for your implementation.